### PR TITLE
integrate tz profiles

### PR DESCRIPTION
**Pull request by Boltman92**

### CODE REVIEW

File: src/components/account-dropdown/account-dropdown-item/account-dropdown-item.tsx [hunk 1]
```diff
@@ -1,11 +1,17 @@
 import React, { FC } from 'react';
 import { Text, View } from 'react-native';
+import FastImage from 'react-native-fast-image';
 
 import { useNetworkInfo } from 'src/hooks/use-network-info.hook';
 import { useContactsSelector } from 'src/store/contact-book/contact-book-selectors';
+import { formatImgUri } from 'src/utils/image.utils';
 
 import { AccountBaseInterface, emptyAccountBase } from '../../../interfaces/account.interface';
```
Comment (keshan3262) [inline src/components/account-dropdown/account-dropdown-item/account-dropdown-item.tsx:9]: Please avoid mixing `src` and relative imports of files from the same levels

File: src/screens/collectibles-home/collectibles-home.tsx [hunk 2]
```diff
@@ -60,15 +67,36 @@ export const CollectiblesHome = () => {
...
 
+  const { alias, twitter, discord, website, github } = selectedAccount.tzProfile || {};
+
+  const socialLinks: SocialLinksInterface[] = [
+    { url: twitter, icon: IconNameEnum.Twitter },
+    { url: isDefined(discord) ? `https://discordapp.com/users/${discord}` : discord, icon: IconNameEnum.Discord },
+    { url: website, icon: IconNameEnum.Website },
+    { url: github, icon: IconNameEnum.Github }
+  ].sort((a, b) => {
+    if (typeof a.url === 'undefined' && typeof b.url === 'undefined') {
```
Comment (keshan3262) [inline src/screens/collectibles-home/collectibles-home.tsx:89]: Try using `isDefined` here

File: src/screens/collectibles-home/collectibles-home.tsx [hunk 3]
```diff
@@ -60,15 +67,36 @@ export const CollectiblesHome = () => {
...
+      return 0;
+    } else if (typeof a.url === 'undefined') {
```
Comment (keshan3262) [inline src/screens/collectibles-home/collectibles-home.tsx:91]: There is no need for `else` keyword after a return statement

File: src/store/wallet/wallet-actions.ts [hunk 4]
```diff
@@ -1,5 +1,7 @@
 import { createAction } from '@reduxjs/toolkit';
 
+import { TzProfile } from 'src/interfaces/tzProfile.interface';
+
 import { AccountInterface } from '../../interfaces/account.interface';
```
Comment (keshan3262) [inline src/store/wallet/wallet-actions.ts:5]: The same as here: https://github.com/madfish-solutions/templewallet-mobile/pull/789/files#diff-dd81f14ebf9edab80c19a772a9ecac5f8f899eb83db3b120a9d029df011baef0R9

File: src/store/wallet/wallet-reducers.ts [hunk 5]
```diff
@@ -1,6 +1,8 @@
 import { createReducer } from '@reduxjs/toolkit';
 import { uniqBy } from 'lodash-es';
 
+import { getSelectedAccount } from 'src/utils/wallet-account-state.utils';
+
 import { VisibilityEnum } from '../../enums/visibility.enum';
```
Comment (keshan3262) [inline src/store/wallet/wallet-reducers.ts:6]: The same

File: ios/Podfile.lock [hunk 6]
```diff
@@ -876,4 +876,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: f8a1b9d32ad4447b452262ee2a33f0ffd98f6d4d
 
-COCOAPODS: 1.12.0
```
Comment (keshan3262) [inline ios/Podfile.lock:879]: I think this file shouldn't have changed

File: src/screens/collectibles-home/collectibles-home.tsx [hunk 7]
```diff
@@ -60,15 +67,36 @@ export const CollectiblesHome = () => {
...
+    { url: website, icon: IconNameEnum.Website },
+    { url: github, icon: IconNameEnum.Github }
+  ].sort((a, b) => {
+    if (isDefined(a.url) && isDefined(b.url)) {
+      return 0;
+    } else if (!isDefined(a.url)) {
+      return 1;
+    } else if (!isDefined(b.url)) {
+      return -1;
+    }
```
Comment (keshan3262) [inline src/screens/collectibles-home/collectibles-home.tsx:95]:
```suggestion
    if (isDefined(a.url) && isDefined(b.url)) {
      return 0;
    }
    if (!isDefined(a.url)) {
      return 1;
    }
    if (!isDefined(b.url)) {
      return -1;
    }
```

**Review by keshan3262 — APPROVED** [review]

File: src/apis/objkt.ts [hunk 8]
```diff
@@ -41,3 +68,17 @@ const buildGetCollectiblesInfoQuery = (address: string) => gql`
     }
   }
 `;
+
+const buildTzProfilesQuery = (address: string) => gql`
+  query MyQuery {
```
Comment (herkoss) [inline src/apis/objkt.ts:73]: Let's find better name for this query

File: src/components/account-dropdown/account-dropdown-item/account-dropdown-item.tsx [hunk 9]
```diff
@@ -34,10 +40,16 @@ export const AccountDropdownItem: FC<AccountDropdownItemProps> = ({
...
   const contacts = useContactsSelector();
   const { metadata } = useNetworkInfo();
+  const selectedAccount = useSelectedAccountSelector();
+  const { logo, alias } = selectedAccount.tzProfile ?? {};
 
   return (
     <View style={styles.root}>
-      <RobotIcon seed={account.publicKeyHash} size={isCollectibleScreen ? COLLECTIBLES_ROBOT_ICON_SIZE : undefined} />
+      {isDefined(logo) && isCollectibleScreen ? (
+        <FastImage style={styles.image} source={{ uri: formatImgUri(selectedAccount.tzProfile?.logo) }} />
```
Comment (herkoss) [inline src/components/account-dropdown/account-dropdown-item/account-dropdown-item.tsx:49]:
```suggestion
        <FastImage style={styles.image} source={{ uri: formatImgUri(logo) }} />
```

**Review by herkoss — APPROVED** [review]

**Pull request merged**